mmv1/templates/terraform/examples/alloydb_cluster_full.tf.erb

```diff
@@ -0,0 +1,78 @@
+resource "google_alloydb_cluster" "<%= ctx[:primary_resource_id] %>" {
+  provider     = google-beta
+  cluster_id   = "<%= ctx[:vars]['alloydb_cluster_name'] %>"
+  location     = "us-central1"
+  network      = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
+
+  initial_user {
+    user     = "<%= ctx[:vars]['alloydb_cluster_name'] %>"
+    password = "<%= ctx[:vars]['alloydb_cluster_name'] %>"
+  }
+
+  automated_backup_policy {
+    location      = "us-central1"
+    backup_window = "1800s"
+    enabled       = true
+
+    encryption_config {
+      kms_key_name = google_kms_crypto_key.crypto_key.id
+    }
+
+    weekly_schedule {
+      days_of_week = ["MONDAY"]
+
+      start_times {
+        hours   = 23
+        minutes = 0
+        seconds = 0
+        nanos   = 0
+      }
+    }
+
+    quantity_based_retention {
+      count = 1
+    }
+
+    labels = {
+      test = "<%= ctx[:vars]['alloydb_cluster_name'] %>"
+    }
+  }
+
+  labels = {
+    test = "<%= ctx[:vars]['alloydb_cluster_name'] %>"
+  }
+
+  encryption_config {
+    kms_key_name = google_kms_crypto_key.crypto_key.id
+  }
+
+  depends_on = [google_project_iam_member.encrypt_role]
+}
+
+resource "google_kms_crypto_key" "crypto_key" {
```

Comment:
kms crypto keys are a complicated resource - they can't be provisioned in normal example-based tests because they can't be deleted. For example usage:

- [TestAccComputeInstance_kmsDiskEncryption](https://github.com/hashicorp/terraform-provider-google/blob/924b5a73a487a472caab55b1a58cb971fcd77a23/google/resource_compute_instance_test.go#L414) (The BootstrapKMSKey function call.)

It looks like this is only needed for the `encryption_config` field - could you leave this field out of the `full` test and instead make sure it's included in a handwritten test? My preference would be to include it in the update test in that case.

Alternately, you could leave `encryption_config` for a later PR.

Reply:
not sure i follow as above full test has passed and im almost certain i did kms test case for another PR in same way (will double check)

Reply:
ah, yeah, I didn't state that well. The test does pass as written; the problem is that it leaks crypto keys because they can't truly be deleted without deleting the entire project - see the note here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_crypto_key

In some cases we work around this by adding a project as part of the test - otherwise, we try to use a bootstrapped key.